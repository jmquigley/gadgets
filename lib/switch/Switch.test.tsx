"use strict";

import {mount, shallow} from "enzyme";
import assert from "power-assert";
import * as React from "react";
import {
	getDefaultSwitchProps,
	getDefaultSwitchState,
	Switch,
	SwitchType
} from "./index";

test("Test retrieval of the Switch props object", () => {
	const props = getDefaultSwitchProps();

	assert(props);
	expect(props).toMatchSnapshot();

	const state = getDefaultSwitchState();

	assert(state);
	expect(state).toMatchSnapshot();
});

test("Test creation of an inny type Switch", () => {
	const ctl = shallow(
		<Switch initialToggle={true} switchType={SwitchType.inny} />
	);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});

test("Test creation of an outy type Switch", () => {
	const ctl = shallow(
		<Switch initialToggle={true} switchType={SwitchType.outy} />
	);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});

test("Test disabling of the Switch control", () => {
	const ctl = mount(<Switch disabled />);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});

test("Test making the Switch control invisible", () => {
	const ctl = mount(<Switch visible={false} />);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});

test("Test the click handler on the Switch control", () => {
	const click = jest.fn();
	const ctl = shallow(<Switch onClick={click} />);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
	assert(ctl.state("toggle") === false);

	const btn = ctl.find(".ui-switch-button");
	assert(btn);

	btn.simulate("click");
	expect(click).toHaveBeenCalled();
	expect(click).toHaveBeenLastCalledWith(true);
	assert(ctl.state("toggle"));

	btn.simulate("click");
	expect(click).toHaveBeenCalledTimes(2);
	expect(click).toHaveBeenLastCalledWith(false);
	assert(ctl.state("toggle") === false);
});
