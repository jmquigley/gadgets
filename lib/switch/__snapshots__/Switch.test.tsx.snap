// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test creation of an inny type Switch 1`] = `
<Wrapper
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  id={null}
  initialToggle={true}
  innyScale={0.6}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  name="Switch"
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  onClick={[Function]}
  onError={[Function]}
  outyScale={1.25}
  padding={null}
  reset={false}
  ripple={false}
  selected={false}
  sizing="normal"
  sliderScale={1.25}
  style={Object {}}
  switchType={0}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <Switch__SliderContainerView
    className="ui-switch"
    height="20px"
    style={Object {}}
  >
    <Switch__SliderView
      backgroundColor={null}
      border={null}
      borderColor={null}
      bottom={null}
      className="ui-switch-slider ui-slider-on"
      color={null}
      controlled={true}
      direction="right"
      disabled={false}
      err={null}
      errorMessage={null}
      focus={false}
      height={null}
      hidden={false}
      id={null}
      initialToggle={true}
      innyScale={0.6}
      left={null}
      location="none"
      maxHeight={null}
      maxWidth={null}
      minHeight={null}
      minWidth={null}
      noborder={false}
      noedit={false}
      nohover={false}
      nopropagation={false}
      notheme={false}
      notooltip={false}
      onClick={[Function]}
      outyScale={1.25}
      padding={null}
      ripple={false}
      selected={false}
      sizing="normal"
      sliderScale={1.25}
      style={
        Object {
          "backgroundColor": "#5cb85c",
        }
      }
      switchType={0}
      testing={true}
      theme={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#c0c0c0",
          "buttonColor": "#666666",
          "chevronColor": "#cccccc",
          "color": "#000000",
          "fontFamily": "OpenSans, Georgia, serif",
          "headerBackgroundColor": "#004358",
          "headerForegroundColor": "#ffffff",
          "headerHoverColor": "#ffe11a",
          "hoverColor": "#c0c0c0",
          "inputBorderColor": "#c0c0c0",
          "itemHoverColor": "#bedb39",
          "outlineColor": "#51cbee",
          "searchFocus": "#51cbee",
          "searchMatch": "#fd7400",
          "selectedBackgroundColor": "#1f8a70",
          "selectedForegroundColor": "#ffe11a",
          "titleBarBackgroundColor": "#004358",
          "titleBarForegroundColor": "#ffe11a",
          "tooltipBackgroundColor": "#708090",
          "tooltipForegroundColor": "#ffffff",
          "transitionDelay": "0.33s",
        }
      }
      tooltip={null}
      top={null}
      visible={true}
      width={null}
      xcss={null}
    >
      <Switch__StyledButton
        className="ui-switch-button"
        height="9.6px"
        onClick={[Function]}
        style={
          Object {
            "left": "5%",
          }
        }
        width="9.6px"
      />
    </Switch__SliderView>
  </Switch__SliderContainerView>
</Wrapper>
`;

exports[`Test creation of an outy type Switch 1`] = `
<Wrapper
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  id={null}
  initialToggle={true}
  innyScale={0.6}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  name="Switch"
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  onClick={[Function]}
  onError={[Function]}
  outyScale={1.25}
  padding={null}
  reset={false}
  ripple={false}
  selected={false}
  sizing="normal"
  sliderScale={1.25}
  style={Object {}}
  switchType={1}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <Switch__SliderContainerView
    className="ui-switch"
    height="20px"
    style={Object {}}
  >
    <Switch__SliderView
      backgroundColor={null}
      border={null}
      borderColor={null}
      bottom={null}
      className="ui-switch-slider ui-slider-on"
      color={null}
      controlled={true}
      direction="right"
      disabled={false}
      err={null}
      errorMessage={null}
      focus={false}
      height={null}
      hidden={false}
      id={null}
      initialToggle={true}
      innyScale={0.6}
      left={null}
      location="none"
      maxHeight={null}
      maxWidth={null}
      minHeight={null}
      minWidth={null}
      noborder={false}
      noedit={false}
      nohover={false}
      nopropagation={false}
      notheme={false}
      notooltip={false}
      onClick={[Function]}
      outyScale={1.25}
      padding={null}
      ripple={false}
      selected={false}
      sizing="normal"
      sliderScale={1.25}
      style={
        Object {
          "backgroundColor": "#5cb85c",
        }
      }
      switchType={1}
      testing={true}
      theme={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#c0c0c0",
          "buttonColor": "#666666",
          "chevronColor": "#cccccc",
          "color": "#000000",
          "fontFamily": "OpenSans, Georgia, serif",
          "headerBackgroundColor": "#004358",
          "headerForegroundColor": "#ffffff",
          "headerHoverColor": "#ffe11a",
          "hoverColor": "#c0c0c0",
          "inputBorderColor": "#c0c0c0",
          "itemHoverColor": "#bedb39",
          "outlineColor": "#51cbee",
          "searchFocus": "#51cbee",
          "searchMatch": "#fd7400",
          "selectedBackgroundColor": "#1f8a70",
          "selectedForegroundColor": "#ffe11a",
          "titleBarBackgroundColor": "#004358",
          "titleBarForegroundColor": "#ffe11a",
          "tooltipBackgroundColor": "#708090",
          "tooltipForegroundColor": "#ffffff",
          "transitionDelay": "0.33s",
        }
      }
      tooltip={null}
      top={null}
      visible={true}
      width={null}
      xcss={null}
    >
      <Switch__StyledButton
        className="ui-switch-button"
        height="20px"
        onClick={[Function]}
        style={
          Object {
            "left": "-15%",
          }
        }
        width="20px"
      />
    </Switch__SliderView>
  </Switch__SliderContainerView>
</Wrapper>
`;

exports[`Test disabling of the Switch control 1`] = `
<Switch
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={true}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  id={null}
  initialToggle={false}
  innyScale={0.6}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  onClick={[Function]}
  outyScale={1.25}
  padding={null}
  ripple={false}
  selected={false}
  sizing="normal"
  sliderScale={1.25}
  style={Object {}}
  switchType={1}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
/>
`;

exports[`Test making the Switch control invisible 1`] = `
<Switch
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  id={null}
  initialToggle={false}
  innyScale={0.6}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  onClick={[Function]}
  outyScale={1.25}
  padding={null}
  ripple={false}
  selected={false}
  sizing="normal"
  sliderScale={1.25}
  style={Object {}}
  switchType={1}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={false}
  width={null}
  xcss={null}
/>
`;

exports[`Test the click handler on the Switch control 1`] = `
<Wrapper
  backgroundColor={null}
  border={null}
  borderColor={null}
  bottom={null}
  className={null}
  color={null}
  controlled={true}
  direction="right"
  disabled={false}
  err={null}
  errorMessage={null}
  focus={false}
  height={null}
  hidden={false}
  id={null}
  initialToggle={false}
  innyScale={0.6}
  left={null}
  location="none"
  maxHeight={null}
  maxWidth={null}
  minHeight={null}
  minWidth={null}
  name="Switch"
  noborder={false}
  noedit={false}
  nohover={false}
  nopropagation={false}
  notheme={false}
  notooltip={false}
  onClick={[MockFunction]}
  onError={[Function]}
  outyScale={1.25}
  padding={null}
  reset={false}
  ripple={false}
  selected={false}
  sizing="normal"
  sliderScale={1.25}
  style={Object {}}
  switchType={1}
  testing={true}
  theme={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#c0c0c0",
      "buttonColor": "#666666",
      "chevronColor": "#cccccc",
      "color": "#000000",
      "fontFamily": "OpenSans, Georgia, serif",
      "headerBackgroundColor": "#004358",
      "headerForegroundColor": "#ffffff",
      "headerHoverColor": "#ffe11a",
      "hoverColor": "#c0c0c0",
      "inputBorderColor": "#c0c0c0",
      "itemHoverColor": "#bedb39",
      "outlineColor": "#51cbee",
      "searchFocus": "#51cbee",
      "searchMatch": "#fd7400",
      "selectedBackgroundColor": "#1f8a70",
      "selectedForegroundColor": "#ffe11a",
      "titleBarBackgroundColor": "#004358",
      "titleBarForegroundColor": "#ffe11a",
      "tooltipBackgroundColor": "#708090",
      "tooltipForegroundColor": "#ffffff",
      "transitionDelay": "0.33s",
    }
  }
  tooltip={null}
  top={null}
  visible={true}
  width={null}
  xcss={null}
>
  <Switch__SliderContainerView
    className="ui-switch"
    height="20px"
    style={Object {}}
  >
    <Switch__SliderView
      backgroundColor={null}
      border={null}
      borderColor={null}
      bottom={null}
      className="ui-switch-slider ui-slider-off"
      color={null}
      controlled={true}
      direction="right"
      disabled={false}
      err={null}
      errorMessage={null}
      focus={false}
      height={null}
      hidden={false}
      id={null}
      initialToggle={false}
      innyScale={0.6}
      left={null}
      location="none"
      maxHeight={null}
      maxWidth={null}
      minHeight={null}
      minWidth={null}
      noborder={false}
      noedit={false}
      nohover={false}
      nopropagation={false}
      notheme={false}
      notooltip={false}
      onClick={[Function]}
      outyScale={1.25}
      padding={null}
      ripple={false}
      selected={false}
      sizing="normal"
      sliderScale={1.25}
      style={
        Object {
          "backgroundColor": "#d9534f",
        }
      }
      switchType={1}
      testing={true}
      theme={
        Object {
          "backgroundColor": "#ffffff",
          "borderColor": "#c0c0c0",
          "buttonColor": "#666666",
          "chevronColor": "#cccccc",
          "color": "#000000",
          "fontFamily": "OpenSans, Georgia, serif",
          "headerBackgroundColor": "#004358",
          "headerForegroundColor": "#ffffff",
          "headerHoverColor": "#ffe11a",
          "hoverColor": "#c0c0c0",
          "inputBorderColor": "#c0c0c0",
          "itemHoverColor": "#bedb39",
          "outlineColor": "#51cbee",
          "searchFocus": "#51cbee",
          "searchMatch": "#fd7400",
          "selectedBackgroundColor": "#1f8a70",
          "selectedForegroundColor": "#ffe11a",
          "titleBarBackgroundColor": "#004358",
          "titleBarForegroundColor": "#ffe11a",
          "tooltipBackgroundColor": "#708090",
          "tooltipForegroundColor": "#ffffff",
          "transitionDelay": "0.33s",
        }
      }
      tooltip={null}
      top={null}
      visible={true}
      width={null}
      xcss={null}
    >
      <Switch__StyledButton
        className="ui-switch-button"
        height="20px"
        onClick={[Function]}
        style={
          Object {
            "left": "45%",
          }
        }
        width="20px"
      />
    </Switch__SliderView>
  </Switch__SliderContainerView>
</Wrapper>
`;
